src/blueprints/tv2/tv2-on-timeline-generate-service.ts

```diff
@@ -281,6 +293,47 @@ export class Tv2OnTimelineGenerateService implements BlueprintOnTimelineGenerate
     }
   }
 
+  private updateTriCasterProgramWithMediaPlayer(timelineObject: Tv2BlueprintTimelineObject, mediaPlayer: Tv2MediaPlayer): void {
+    if (timelineObject.content.deviceType !== DeviceType.TRICASTER || timelineObject.content.type !== TriCasterType.ME) {
+      return
+    }
+    const triCasterMeTimelineObject: TriCasterMixEffectTimelineObject = timelineObject as TriCasterMixEffectTimelineObject
+    if (triCasterMeTimelineObject.content.me.type !== TriCasterMixEffectContentType.PROGRAM) {
+      return
+    }
+    triCasterMeTimelineObject.content.me.programInput = this.prefixSourceInputWithTriCasterPrefix(mediaPlayer.videoMixerSource)
+  }
+
+  private prefixSourceInputWithTriCasterPrefix(sourceInput: number): TriCasterInputName {
+    return `input${sourceInput}`
+  }
+
+  private updateTriCasterLookaheadWithMediaPlayer(timelineObject: Tv2BlueprintTimelineObject, mediaPlayer: Tv2MediaPlayer): void {
+    if (timelineObject.content.deviceType !== DeviceType.TRICASTER || timelineObject.content.type !== TriCasterType.MIX_OUTPUT) {
+      return
+    }
+    const triCasterMixOutputTimelineObject: TriCasterMixOutputTimelineObject = timelineObject as TriCasterMixOutputTimelineObject
+    triCasterMixOutputTimelineObject.content.source = this.prefixSourceInputWithTriCasterPrefix(mediaPlayer.videoMixerSource)
+  }
+
+  private updateTriCasterSplitScreenBoxesWithMediaPlayer(timelineObject: Tv2BlueprintTimelineObject, mediaPlayer: Tv2MediaPlayer): void {
+    if (timelineObject.content.deviceType !== DeviceType.TRICASTER || timelineObject.content.type !== TriCasterType.ME) {
+      return
+    }
+    const triCasterMeTimelineObject: TriCasterMixEffectTimelineObject = timelineObject as TriCasterMixEffectTimelineObject
+    if (triCasterMeTimelineObject.content.me.type !== TriCasterMixEffectContentType.EFFECT_MODE || triCasterMeTimelineObject.content.me.transitionEffect !== TriCasterTransition.SPLIT_SCREEN) {
+      return
+    }
+    const meEffectContent: TriCasterMixEffectEffectModeContent = triCasterMeTimelineObject.content.me
+    for (const key of Object.keys(meEffectContent.layers)) {
+      const layer: TriCasterLayer | undefined = meEffectContent.layers[key as TriCasterLayerName]
+      if (!layer || !layer.input || layer.input !== this.prefixSourceInputWithTriCasterPrefix(A_B_SOURCE_INPUT_PLACEHOLDER)) {
+        continue
+      }
+      layer.input = this.prefixSourceInputWithTriCasterPrefix(mediaPlayer.videoMixerSource)
+    }
```

Comment:
The key is only used to extract the layer value. A level of abstraction can be removed by using `Object.values` to work directly with the values or `Array.map` to iterate over the list. E.g.
```typescript
  const tricasterABSourceInputPlaceholder: string = this.prefixSourceInputWithTriCasterPrefix(A_B_SOURCE_INPUT_PLACEHOLDER)
  meEffectContent.layers = meEffectContent.layers
    .map(layer => {
      if (layer.input !== tricasterABSourceInputPlaceholder) {
        return layer
      }
      return { ...layer, input: this.prefixSourceInputWithTriCasterPrefix(mediaPlayer.videoMixerSource) }
    })
```
or
```typescript
  const tricasterABSourceInputPlaceholder: string = this.prefixSourceInputWithTriCasterPrefix(A_B_SOURCE_INPUT_PLACEHOLDER)
  for (const layer of Object.values(meEffectContent.layers)) {
     if (layer.input !== tricasterABSourceInputPlaceholder) {
       continue
     }
     layer.input = this.prefixSourceInputWithTriCasterPrefix(mediaPlayer.videoMixerSource)
  }

Reply:
Updated to your latter suggestion.